tasks/RedHat/generic/assert-process-resource-limits.yml

```diff
@@ -6,156 +6,181 @@
 - name: Get info about file /etc/security/limits.d/99-sap.conf
   stat:
     path: /etc/security/limits.d/99-sap.conf
-  register: stat_sap_conf_2
+  register: __stat_limits_sap_conf
 
 - name: Assert that file /etc/security/limits.d/99-sap.conf exists
   assert:
-    that: "stat_sap_conf_2.stat.isreg == true"
-    fail_msg: "FAIL: The file /etc/security/limits.d/99-sap.conf does not exist or is no regular file!"
-    success_msg: "PASS: The file /etc/security/limits.d/99-sap.conf exists and is a regular file."
+    that: __stat_limits_sap_conf.stat.exists
+    fail_msg: "FAIL: File /etc/security/limits.d/99-sap.conf does not exist!"
+    success_msg: "PASS: File /etc/security/limits.d/99-sap.conf exist."
   ignore_errors: "{{ sap_preconfigure_assert_ignore_errors|d(false) }}"
 
+- name: Assert that file /etc/security/limits.d/99-sap.conf is a regular file
+  assert:
+    that: __stat_limits_sap_conf.stat.isreg
+    fail_msg: "FAIL: File /etc/security/limits.d/99-sap.conf is not a regular file!"
+    success_msg: "PASS: File /etc/security/limits.d/99-sap.conf is a regular file."
+  ignore_errors: "{{ sap_preconfigure_assert_ignore_errors|d(false) }}"
+  when: __stat_limits_sap_conf.stat.exists
+
 - name: Check if the hard limit of nofile for group sapsys is 65536
   command: awk '/sapsys/&&/nofile/&&/hard/{print $NF}' /etc/security/limits.d/99-sap.conf
-  register: awk_nofile_hard_result
+  register: __sap_conf_nofile_hard_result
   changed_when: no
   ignore_errors: yes
-  when: stat_sap_conf_2.stat.isreg
+  when:
+    - __stat_limits_sap_conf.stat.exists
+    - __stat_limits_sap_conf.stat.isreg
```

Comment:
ok - but I am curious if you have ever seen a case where `__stat_limits_sap_conf.stat.isreg` is `true` but `__stat_limits_sap_conf.stat.exists` is `false`?

Reply:
could be a link...